hasktorch/src/Torch/Typed/NN/Recurrent/LSTM.hs

```diff
@@ -0,0 +1,145 @@
+{-# LANGUAGE AllowAmbiguousTypes     #-}
+{-# LANGUAGE ConstraintKinds         #-}
+{-# LANGUAGE DataKinds               #-}
+{-# LANGUAGE DeriveGeneric           #-}
+{-# LANGUAGE FlexibleContexts        #-}
+{-# LANGUAGE FlexibleInstances       #-}
+{-# LANGUAGE GADTs                   #-}
+{-# LANGUAGE MultiParamTypeClasses   #-}
+{-# LANGUAGE NoStarIsType            #-}
+{-# LANGUAGE OverloadedLists         #-}
+{-# LANGUAGE PolyKinds               #-}
+{-# LANGUAGE RankNTypes              #-}
+{-# LANGUAGE RecordWildCards         #-}
+{-# LANGUAGE ScopedTypeVariables     #-}
+{-# LANGUAGE TypeApplications        #-}
+{-# LANGUAGE TypeFamilies            #-}
+{-# LANGUAGE TypeOperators           #-}
+{-# LANGUAGE UndecidableInstances    #-}
+{-# LANGUAGE UndecidableSuperClasses #-}
+{-# OPTIONS_GHC -fplugin GHC.TypeLits.Normalise #-}
+{-# OPTIONS_GHC -fplugin GHC.TypeLits.KnownNat.Solver #-}
+{-# OPTIONS_GHC -fplugin GHC.TypeLits.Extra.Solver #-}
+{-# OPTIONS_GHC -fconstraint-solver-iterations=0 #-}
+
+module Torch.Typed.NN.Recurrent.LSTM where
+
+import           Control.Monad                  ( foldM
+                                                , when
+                                                , void
+                                                )
+import           Control.Exception.Safe         ( try
+                                                , SomeException(..)
+                                                )
+import           Data.List                      ( foldl'
+                                                , intersperse
+                                                , scanl'
+                                                )
+import           Data.Maybe                     ( catMaybes )
+import           Data.Reflection
+import           Foreign.ForeignPtr
+import           GHC.Generics
+import           GHC.TypeLits
+import           GHC.TypeLits.Extra
+import           System.Environment
+import           System.IO.Unsafe
+import           Prelude                 hiding ( tanh )
+import qualified ATen.Cast                     as ATen
+import qualified ATen.Class                    as ATen
+import qualified ATen.Type                     as ATen
+import qualified ATen.Managed.Type.Tensor      as ATen
+import qualified Torch.Autograd                as A
+import qualified Torch.DType                   as D
+import qualified Torch.Functions               as D
+import qualified Torch.NN                      as A
+import qualified Torch.Tensor                  as D
+import qualified Torch.TensorFactories         as D
+import           Torch.Typed
+import           Torch.Typed.Factories
+import           Torch.Typed.Native      hiding ( linear )
+import           Torch.Typed.NN
+
+
+
+data LSTMCellSpec (dtype :: D.DType) (inputDim :: Nat) (hiddenDim :: Nat) =
+    LSTMCellRandomInitState -- ^ Random initial memory cells and hiddenstate values
```

Comment:
"Random state" is very uninformative. We should either mention what's the distribution, or better e.g. let people provide a tensor constructor (something like `forall shape. Tensor shape`).

Reply:
I concur, but I didn't see any established convention for initialization states.  My Preference would be something like a reference to a canonical initialization in the literature, but that might raise the bar for contributions.  I defer the maintainers on the best convention.